src/Product/SyncerHooks.php

```diff
@@ -66,56 +70,71 @@ class SyncerHooks implements Service, Registerable {
 	 */
 	protected $merchant_center;
 
+	/**
+	 * @var WC
+	 */
+	protected $wc;
+
 	/**
 	 * SyncerHooks constructor.
 	 *
 	 * @param BatchProductHelper    $batch_helper
 	 * @param ProductHelper         $product_helper
 	 * @param JobRepository         $job_repository
 	 * @param MerchantCenterService $merchant_center
+	 * @param WC                    $wc
 	 */
 	public function __construct(
 		BatchProductHelper $batch_helper,
 		ProductHelper $product_helper,
 		JobRepository $job_repository,
-		MerchantCenterService $merchant_center
+		MerchantCenterService $merchant_center,
+		WC $wc
 	) {
 		$this->batch_helper        = $batch_helper;
 		$this->product_helper      = $product_helper;
 		$this->update_products_job = $job_repository->get( UpdateProducts::class );
 		$this->delete_products_job = $job_repository->get( DeleteProducts::class );
 		$this->merchant_center     = $merchant_center;
+		$this->wc                  = $wc;
 	}
 
 	/**
 	 * Register a service.
 	 */
 	public function register(): void {
-		// only register the hooks if Merchant Center is set up.
-		if ( ! $this->merchant_center->is_setup_complete() ) {
+		// only register the hooks if Merchant Center is set up and connected.
+		if ( ! $this->merchant_center->is_connected() ) {
 			return;
 		}
 
-		$update = function( int $product_id ) {
-			$this->handle_update_product( $product_id );
+		$update_by_object = function ( int $product_id, WC_Product $product ) {
```

Comment:
We have a mix in our code of `function(` and `function (`. If I recall correctly from a previous discussion `function(` without space was preferred. But either way I think it's still personal preference.